ORA-00001: unique constraint (UT3.UT_COVERAGE_SOURCES_TMP_PK) violated

**Describe the bug**
A clear and concise description of what the bug is.
When running utPLSQL on a PDB and user running the tests has access to DBA_SOURCE and the scope of coverage is set to include SYS or MDSYS schema, the test runs will fail becayse package bodies exist in both CDB and PDB and DBA_SOURCE shows one version for ORIGIN_CON_ID = 0 while another version for ORIGIN_CON_ID = SYS_CONTYEXT('USERENV','CON_ID')

So a duplicate source line is found and coverage reporting will fail with:
```
[debug] Java API Version = 3.2.3
[info] Enabled dbms_output.
[info] utPLSQL Version = v3.2.01.4373
[info] Invoking TestRunner with: /home/runner/work/utPLSQL-maven-plugin/utPLSQL-maven-plugin/src/test/target
[info] Writing report UT_DOCUMENTATION_REPORTER to Console
plsql
  examples
    TEST_PKG_TEST_ME
      Checking if function ('1') returns 1 [.005 sec]
      Checking if function ('0') returns 0 [.002 sec]
      Checking if function (NULL) returns NULL [.002 sec]
      Checking if procedure (NULL) insert [.004 sec]
      Checking if procedure (NOT NULL) insert [.006 sec]
      Checking if procedure (NOT NULL) insert while existing [.002 sec]
      Demonstrating the use of cursor [.015 sec]
 
 
Warnings:
 
  1) plsql.examples.test_pkg_test_me
      Unable to perform automatic rollback after test suite. An implicit or explicit commit/rollback occurred in procedures:
        app.test_pkg_test_me.test_pr_test_me_not_null
        app.test_pkg_test_me.test_pr_test_me_exists
        app.test_pkg_test_me.test_pr_test_me_cursor
      Use the "--%rollback(manual)" annotation or remove commit/rollback/ddl statements that are causing the issue.
	at org.utplsql.api.testRunner.DynamicTestRunnerStatement.execute(DynamicTestRunnerStatement.java:90)
	at org.utplsql.api.TestRunner.run(TestRunner.java:245)
	at org.utplsql.maven.plugin.UtPlsqlMojo.execute(UtPlsqlMojo.java:210)
	... 3 more
Caused by: Error : 1, Position : 0, SQL = BEGIN ut_runner.run(a_paths => :1 , a_reporters => :2 , a_color_console => (case :3 when 1 then true else false end), a_fail_on_errors => (case :4 when 1 then true else false end), a_client_character_set => :5 , a_random_test_order => (case :6 when 1 then true else false end), a_include_schema_expr => :7 );END;, Original SQL = BEGIN ut_runner.run(a_paths => ?, a_reporters => ?, a_color_console => (case ? when 1 then true else false end), a_fail_on_errors => (case ? when 1 then true else false end), a_client_character_set => ?, a_random_test_order => (case ? when 1 then true else false end), a_include_schema_expr => ?);END;, Error Message = ORA-00001: unique constraint (UT3.UT_COVERAGE_SOURCES_TMP_PK) violated
ORA-06512: at "UT3.UT_RUNNER", line 151
ORA-06512: at "UT3.UT_COVERAGE_HELPER", line 37
ORA-06512: at "UT3.UT_COVERAGE", line 212
ORA-06512: at "UT3.UT_COVERAGE", line 271
ORA-06512: at "UT3.UT_COVERAGE_SONAR_REPORTER", line 95
ORA-06512: at "UT3.UT_REPORTER_BASE", line 195
ORA-06512: at "UT3.UT_EVENT_MANAGER", line 70
ORA-06512: at "UT3.UT_EVENT_MANAGER", line 80
ORA-06512: at "UT3.UT_RUN", line 66
ORA-06512: at "UT3.UT_SUITE_ITEM", line 50
ORA-06512: at "UT3.UT_RUNNER", line 144
ORA-06512: at line 1

	at oracle.jdbc.driver.T4CTTIoer11.processError(T4CTTIoer11.java:723)
	... 20 more

[INFO] 
[INFO] Results:
[INFO] 
Error:  Errors: 
Error:    UtPlsqlMojoTest.include_object_expr:371 » MojoExecution ORA-00001: unique constraint (UT3.UT_COVERAGE_SOURCES_TMP_PK) violated
ORA-06512: at "UT3.UT_RUNNER", line 151
ORA-06512: at "UT3.UT_COVERAGE_HELPER", line 37
ORA-06512: at "UT3.UT_COVERAGE", line 212
ORA-06512: at "UT3.UT_COVERAGE", line 271
ORA-06512: at "UT3.UT_COVERAGE_SONAR_REPORTER", line 95
ORA-06512: at "UT3.UT_REPORTER_BASE", line 195
ORA-06512: at "UT3.UT_EVENT_MANAGER", line 70
ORA-06512: at "UT3.UT_EVENT_MANAGER", line 80
ORA-06512: at "UT3.UT_RUN", line 66
ORA-06512: at "UT3.UT_SUITE_ITEM", line 50
ORA-06512: at "UT3.UT_RUNNER", line 144
ORA-06512: at line 1

https://docs.oracle.com/error-help/db/ora-00001/
Error:    UtPlsqlMojoTest.include_schema_expr:406 » MojoExecution ORA-00001: unique constraint (UT3.UT_COVERAGE_SOURCES_TMP_PK) violated
ORA-06512: at "UT3.UT_RUNNER", line 151
ORA-06512: at "UT3.UT_COVERAGE_HELPER", line 37
ORA-06512: at "UT3.UT_COVERAGE", line 212
ORA-06512: at "UT3.UT_COVERAGE", line 271
ORA-06512: at "UT3.UT_COVERAGE_SONAR_REPORTER", line 95
ORA-06512: at "UT3.UT_REPORTER_BASE", line 195
ORA-06512: at "UT3.UT_EVENT_MANAGER", line 70
ORA-06512: at "UT3.UT_EVENT_MANAGER", line 80
ORA-06512: at "UT3.UT_RUN", line 66
ORA-06512: at "UT3.UT_SUITE_ITEM", line 50
ORA-06512: at "UT3.UT_RUNNER", line 144
ORA-06512: at line 1

https://docs.oracle.com/error-help/db/ora-00001/
[INFO] 
Error:  Tests run: 31, Failures: 0, Errors: 2, Skipped: 0
[INFO] 
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] Skipping utPLSQL Maven Plugin
[INFO] This project has been banned from the build due to previous failures.
[INFO] ------------------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  35.517 s
[INFO] Finished at: 2026-04-23T15:50:37Z
[INFO] ------------------------------------------------------------------------
Error:  Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.5.5:test (default-test) on project utplsql-maven-plugin: 
Error:  
Error:  See /home/runner/work/utPLSQL-maven-plugin/utPLSQL-maven-plugin/target/surefire-reports for the individual test results.
Error:  See dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
Error:  -> [Help 1]
Error:  
Error:  To see the full stack trace of the errors, re-run Maven with the -e switch.
Error:  Re-run Maven using the -X switch to enable full debug logging.
Error:  
Error:  For more information about the errors and possible solutions, please read the following articles:
Error:  [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
Error: Process completed with exit code 1.
```
You can see this in test runs here:
https://github.com/utPLSQL/utPLSQL-maven-plugin/actions/runs/24844825275/job/72729154909 

**Provide version info**
Information about utPLSQL and Database version, 

[debug] Java API Version = 3.2.3
[info] Enabled dbms_output.
[info] utPLSQL Version = v3.2.01.4373

DB: Oracle 21 Free 
Image: gvenzl/oracle-xe:21-slim

**Information about client software**
Azure Pipeline

**To Reproduce**
Run mvn clean verify for the utplql maven plugin project. 

**Expected behavior**
utPLSQL should not read DBA_SOURCE without filtering CON_ID.
A filter `  AND ORIGIN_CON_ID = SYS_CONTEXT('USERENV', 'CON_ID')` should be added to queries on DBA_SOURCE


**Example code**
```sql
create user APP identified by pass quota unlimited on USERS default tablespace USERS;
grant create session, create procedure, create type, create table, create sequence, create view to APP;
grant select any dictionary to APP;

-- auto-generated definition
create table TO_TEST_ME
(
    SNAME VARCHAR2(10)
)
/

create PACKAGE PKG_TEST_ME AS
    FUNCTION FC_TEST_ME(PPARAM1 IN VARCHAR2) RETURN NUMBER;
    PROCEDURE PR_TEST_ME(PSNAME IN VARCHAR2);
END PKG_TEST_ME;
/
create PACKAGE BODY PKG_TEST_ME IS
    --
    -- This
    --
    FUNCTION FC_TEST_ME(PPARAM1 IN VARCHAR2) RETURN NUMBER IS
    BEGIN
        IF PPARAM1 IS NULL THEN
            RETURN NULL;
        ELSIF PPARAM1 = '1' THEN
            RETURN 1;
        ELSE
            RETURN 0;
        END IF;
    END FC_TEST_ME;

    PROCEDURE PR_TEST_ME(PSNAME IN VARCHAR2) IS
    BEGIN
        IF PSNAME IS NULL THEN
            NULL;
        ELSE
            INSERT INTO TO_TEST_ME (SNAME) VALUES (PSNAME);
            COMMIT;
        END IF;
    END PR_TEST_ME;

END PKG_TEST_ME;
/

connect APP/pass;

begin
    ut.run('APP', a_reporter=> UT_COVERALLS_REPORTER(), a_include_schema_expr =>'*');
end;
/

```

Results in error:
```
ORA-00001: unique constraint (UT3.UT_COVERAGE_SOURCES_TMP_PK) violated on table UT3.UT_COVERAGE_SOURCES_TMP columns (OWNER, NAME, TYPE, LINE)
ORA-06512: at "UT3.UT_RUNNER", line 151
ORA-03301: (ORA-00001 details) row with column values (OWNER:'MDSYS', NAME:'SDO_OLS_DIRECTORY', TYPE:'PACKAGE BODY', LINE:1) already exists
ORA-06512: at "UT3.UT_COVERAGE_HELPER", line 37
ORA-06512: at "UT3.UT_COVERAGE", line 212
ORA-06512: at "UT3.UT_COVERAGE", line 271
ORA-06512: at "UT3.UT_COVERALLS_REPORTER", line 89
ORA-06512: at "UT3.UT_REPORTER_BASE", line 195
ORA-06512: at "UT3.UT_EVENT_MANAGER", line 70
ORA-06512: at "UT3.UT_EVENT_MANAGER", line 80
ORA-06512: at "UT3.UT_RUN", line 66
ORA-06512: at "UT3.UT_SUITE_ITEM", line 50
ORA-06512: at "UT3.UT_RUNNER", line 144
ORA-06512: at "UT3.UT", line 143
ORA-06512: at "UT3.UT", line 577
ORA-06512: at "UT3.UT", line 744
ORA-06512: at line 2


```

**Additional context**
Add any other context about the problem here.

--- a/source/core/annotations/ut_annotation_manager.pkb
+++ b/source/core/annotations/ut_annotation_manager.pkb
@@ -121,6 +121,7 @@ create or replace package body ut_annotation_manager as
                    and s.type  = r.object_type
                  where s.owner = ']'||ut_utils.qualified_sql_name(a_object_owner)||q'['
                    and s.type  = ']'||ut_utils.qualified_sql_name(a_object_type)||q'['
+                   and s.origin_con_id = sys_context('userenv', 'con_id')
                ) x
           where x.is_annotated = 'Y'
           order by x.name, x.line]'
@@ -268,6 +269,7 @@ create or replace package body ut_annotation_manager as
             where s.type  = :a_object_type
               and s.owner = :a_object_owner
               and s.name  = :a_object_name
+              and s.origin_con_id = sys_context('userenv', 'con_id')
           order by s.line]'
         using a_object_name, a_object_type, a_object_owner, a_object_name;
       return l_result;

--- a/source/core/coverage/ut_coverage.pkb
+++ b/source/core/coverage/ut_coverage.pkb
@@ -59,7 +59,7 @@ create or replace package body ut_coverage is
            {filters}
            {regex_exc_filters}
            and not exists (
-              select /*+ cardinality(el {excuded_objects_cardinality})*/ 1
+              select /*+ cardinality(el {excluded_objects_cardinality})*/ 1
                 from table(:l_excluded_objects) el
                where s.owner = el.owner and  s.name = el.name
            )
@@ -153,7 +153,7 @@ create or replace package body ut_coverage is
     l_result := replace(l_result, '{filters}',              l_filters);
     l_result := replace(l_result, '{mappings_cardinality}', l_mappings_cardinality);
     l_result := replace(l_result, '{skipped_objects_cardinality}', ut_utils.scale_cardinality(cardinality(a_skip_objects)));
-    l_result := replace(l_result, '{excuded_objects_cardinality}', ut_utils.scale_cardinality(cardinality(coalesce(a_coverage_options.exclude_objects, ut_object_names()))));
+    l_result := replace(l_result, '{excluded_objects_cardinality}', ut_utils.scale_cardinality(cardinality(coalesce(a_coverage_options.exclude_objects, ut_object_names()))));
     l_result := replace(l_result, '{regex_exc_filters}', l_regex_exc_filters);
 
     return l_result;

--- a/source/core/coverage/ut_coverage.pkb
+++ b/source/core/coverage/ut_coverage.pkb
@@ -55,6 +55,7 @@ create or replace package body ut_coverage is
                s.text
           from {sources_view} s {join_file_mappings}
          where s.type in ('PACKAGE BODY', 'TYPE BODY', 'PROCEDURE', 'FUNCTION', 'TRIGGER')
+           and s.origin_con_id = sys_context('userenv', 'con_id')
            {filters}
            {regex_exc_filters}
            and not exists (

--- a/test/ut3_user/reporters/test_coverage/test_proftab_coverage.pkb
+++ b/test/ut3_user/reporters/test_coverage/test_proftab_coverage.pkb
@@ -207,5 +207,29 @@ create or replace package body test_proftab_coverage is
     ut.expect(l_actual).to_be_like(coverage_helper.substitute_covered_package(l_expected));
   end;
 
+  procedure no_issue_with_duplicate_source is
+    l_expected  clob;
+    l_actual    clob;
+  begin
+    --Arrange
+    l_expected := '%<file path="package body mdsys.sdo_util%">%';
+    --Act
+    l_actual :=
+      ut3_tester_helper.coverage_helper.run_tests_as_job(
+        coverage_helper.substitute_covered_package(
+            q'[
+              ut3_develop.ut.run(
+                'ut3_develop.test_dummy_coverage.zero_coverage',
+                ut3_develop.ut_coverage_sonar_reporter(),
+                a_include_objects => ut3_develop.ut_varchar2_list('MDSYS.SDO_UTIL')
+              )
+            ]'
+          )
+        );
+
+    --Running above line is successful if there is no exception thrown due to duplicate lines in coverage
+    ut.expect(l_actual).to_be_like(l_expected);
+  end;
+
 end;
 /

--- a/test/ut3_user/reporters/test_coverage/test_proftab_coverage.pkb
+++ b/test/ut3_user/reporters/test_coverage/test_proftab_coverage.pkb
@@ -212,7 +212,7 @@ create or replace package body test_proftab_coverage is
     l_actual    clob;
   begin
     --Arrange
-    l_expected := '%<file path="package body mdsys.sdo_util%">%';
+    l_expected := '%<file path="trigger mdsys.%">%';
     --Act
     l_actual :=
       ut3_tester_helper.coverage_helper.run_tests_as_job(
@@ -221,7 +221,7 @@ create or replace package body test_proftab_coverage is
               ut3_develop.ut.run(
                 'ut3_develop.test_dummy_coverage.zero_coverage',
                 ut3_develop.ut_coverage_sonar_reporter(),
-                a_include_objects => ut3_develop.ut_varchar2_list('MDSYS.SDO_UTIL')
+                a_include_schema_expr =>'MDSYS'
               )
             ]'
           )

--- a/test/ut3_user/reporters/test_coverage/test_proftab_coverage.pkb
+++ b/test/ut3_user/reporters/test_coverage/test_proftab_coverage.pkb
@@ -208,11 +208,8 @@ create or replace package body test_proftab_coverage is
   end;
 
   procedure no_issue_with_duplicate_source is
-    l_expected  clob;
     l_actual    clob;
   begin
-    --Arrange
-    l_expected := '%<file path="trigger mdsys.%">%';
     --Act
     l_actual :=
       ut3_tester_helper.coverage_helper.run_tests_as_job(
@@ -221,14 +218,14 @@ create or replace package body test_proftab_coverage is
               ut3_develop.ut.run(
                 'ut3_develop.test_dummy_coverage.zero_coverage',
                 ut3_develop.ut_coverage_sonar_reporter(),
-                a_include_schema_expr =>'MDSYS'
+                a_include_schema_expr =>'SYS'
               )
             ]'
           )
         );
 
     --Running above line is successful if there is no exception thrown due to duplicate lines in coverage
-    ut.expect(l_actual).to_be_like(l_expected);
+    ut.expect(l_actual).not_to_be_null();
   end;
 
 end;

--- a/test/ut3_user/reporters/test_coverage/test_proftab_coverage.pks
+++ b/test/ut3_user/reporters/test_coverage/test_proftab_coverage.pks
@@ -28,5 +28,8 @@ create or replace package test_proftab_coverage is
   --%test(reports zero coverage on each line of non-executed database object - Issue #917)
   procedure report_zero_coverage;
 
+  --%test(Allows reporting coverage on objects across SYS or MDSYS without errors - Issue #1323 )
+  procedure no_issue_with_duplicate_source;
+
 end;
 /

